Fix for hidden avatars near comments on small screens 883

sirrius-a-b opened the pull request:
Fixes # .
- for issue #883 

Changes:
- un-hide avatars in the comments on smaller screens by
- setting overflow:unset; for .Comment__text on small screens

jm90m commented:
Hey thanks for the contribution :D!

So this might not work... as it is related to this issue right here: https://github.com/busyorg/busy/issues/771

The `overflow: hidden;` was added to fix comments with code blocks from not scrolling. 

Test out your fix here: 

`test/@hellosteem/5ihr1p-test#@hellosteem/re-hellosteem-5ihr1p-test-20171018t132820462z`

and look into the comments with codeblocks, the scroll won't appear all the way with this.

sirrius-a-b commented:
I see, well, the `overflow:hidden;` does prevent those comments from scrolling but they get cut and are behind the right menu. I think #771 should maybe be reopened.

This can be better seen in the screen shot below. Notice how long the comment `code` tag actually is when I highlight it:
![comments-behind-menu-small](https://user-images.githubusercontent.com/32915194/32290566-c31aca88-bf3a-11e7-8f58-3f58a36e6df8.png)

sirrius-a-b commented:
I am looking for a fix to satisfy both cases.

sirrius-a-b commented:
Now it is fixed, the comments do not overflow or scroll and the avatars are showing up. Give it a good test though and let me know if anything.

jm90m commented:
@bonustrack can we get a deploy on this so we can test?

bonustrack commented:
Yes, on the way

jm90m commented:
@sirrius-a-b So your current fix removes the scrollbar in comment code blocks, we want to retain the scrollbar. 

<img width="254" alt="screen shot 2017-11-01 at 3 55 23 pm" src="https://user-images.githubusercontent.com/5067716/32294646-205aea84-bf1d-11e7-85ad-1c5cda182413.png">

jm90m commented on src/components/Comments/Comment.less:
```diff
@@ -11,6 +11,14 @@
 }
 
+.Comments > .Comment {
+    overflow:hidden;
```
should be `overflow: hidden` with a space

jm90m commented on src/components/Comments/Comment.less:
```diff
@@ -11,6 +11,14 @@
+}
+
+.Comments > .Comment > .Comment__text {
+    width:calc(~'100% - 52px');
```
this seems to work, but I'm not too sure if hardcoded values would be right.

sirrius-a-b replied:
It is `52px` because the zero level comment has `.Avatar` set at `40px` and `.Comment__text` has `margin-left:12px;` , together making 52. I will look on how to not be hardcoded.

jm90m replied:
thanks! great fix though :)

jm90m commented on src/components/Comments/Comment.less:
```diff
@@ -50,9 +58,14 @@
     position: relative;
     top: -4px;
-    width: 100%;
+    width: calc(~'100% - 44px');
```
it might be good to put this in `common.less` and accompany it with the avatar width:

```
@default-avatar-size: 40px;
@comment-text-offset: @default-avatar-size + 4px;
```

But see in `https://github.com/busyorg/busy/blob/new-design/src/components/Comments/Comment.js#L252` ... avatar size can be 40px or 32px, and I think default avatar in the `Avatar.js` component is 34px;

sirrius-a-b replied:
Should I not better put in in `custom.less` instead of `common.less` ? I see that in `custom.less` are most of the variables. Please advise.

jm90m replied:
yeah thats what i meant sorry, `custom.less`

sirrius-a-b replied:
Thanks, I will update it shortly.

sirrius-a-b commented:
I implemented all @jm90m recommendations to make it not be hard coded by adding vars for comments in `custom.less` and used them in `Comment.less` . I also added a space after the `:` on a few occasions as it has been suggested. I will remember this code convention now.

jm90m commented on src/styles/custom.less:
```diff
@@ -55,3 +55,10 @@
 @layout-zero-trigger-height : 42px;
+
+// Comments
+@comment-text-left-margin: 12px;
```
This is great, thanks for providing more context on what the values actually mean. @bonustrack what do you think of this fix? It seems to be working for me.
